configuration/configuration.go

```diff
@@ -109,3 +114,56 @@ func ExtractBlob(uri string) (blob []byte, err error) {
 	}
 	return blob, nil
 }
+
+// Save stores the payloads.Configure struct passed in 'conf'
+// using the backend defined by the scheme in the configuration URI.
+// Previous to save configuration, backup is retrieved from the URI
+// specified by the backup variable and saved in "backup.YYYYMMDD_HHMMSS"
+func Save(conf *payloads.Configure, backup string) error {
+	var d driver
+
+	if conf == nil {
+		return errors.New("unable to store invalid configuration")
+	}
+
+	// create a backup first
+	driverType, err := discoverDriver(backup)
+	if err != nil {
+		return err
+	}
+	switch driverType {
+	case payloads.Filesystem:
+		d = &file{}
+	}
+	pyld, err := d.fetchConfiguration(backup)
+	if err != nil {
+		return fmt.Errorf("unable to retrieve current configuration for backup: %v", err)
+	}
+
+	uri := pyld.Configure.Scheduler.ConfigStorageURI
```

Comment:
if storageURI is changed, a backup is written in `currentStorateURI`.[timeFmt](https://github.com/mrkz/ciao/blob/77cb6528f281ac291f6f47a69e5ec25653e18893/configuration/configuration.go#L30) filename (e.g: `file:///etc/ciao/configuration.yaml.20060102_150405` ) and the new configuration is written in the storageURI defined in new configuration (if storageURI wasn't changed, it would write new conf in `file:///etc/ciao/configuration.yaml`)